pim-tool-back/src/main/java/pilotproject/Project_Management_ELCA/service/impl/ProjectServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package pilotproject.Project_Management_ELCA.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.orm.ObjectOptimisticLockingFailureException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pilotproject.Project_Management_ELCA.model.dto.ProjectDto;
+import pilotproject.Project_Management_ELCA.model.entity.Project;
+import pilotproject.Project_Management_ELCA.model.exception.ConcurrentUpdateException;
+import pilotproject.Project_Management_ELCA.model.exception.ProjectNumberExistedException;
+import pilotproject.Project_Management_ELCA.repository.GroupRepository;
+import pilotproject.Project_Management_ELCA.repository.ProjectRepository;
+import pilotproject.Project_Management_ELCA.service.ProjectService;
+import pilotproject.Project_Management_ELCA.util.ApplicationMapper;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+@Service
+@Transactional
+public class ProjectServiceImpl implements ProjectService {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    private final ProjectRepository projectRepository;
+    @Autowired
+    private ApplicationMapper mapper;
```

Comment:
[N] The mapping should occur in Controller instead of Service, according to the architecture of this project.